geowebcache/core/src/main/java/org/geowebcache/storage/blobstore/file/FileBlobStore.java

```diff
@@ -696,15 +789,108 @@ private Properties getLayerMetadata(final String layerName) {
+        final String lockObj = metadataFile.getAbsolutePath().intern();
+        synchronized (lockObj) {
+            if (metadataFile.exists()) {
+                FileInputStream in;
+                GZIPInputStream gzipIn;
+                try {
+                    in = new FileInputStream(metadataFile);
+                    gzipIn = new GZIPInputStream(in);
+                } catch (IOException e) {
+                    throw new UncheckedIOException(e);
+                }
+                try {
+                	int tryagain = 0;
+                    while (tryagain < FileBlobStore.METADATA_MAX_RW_ATTEMPTS) {
+                        // Read current metadatafile date modification (currentModTime)
+                        long currentModDate = metadataFile.lastModified();
+	                	if (lastModified == currentModDate) {
+	                		properties.load(gzipIn);
+	                		tryagain = FileBlobStore.METADATA_MAX_RW_ATTEMPTS;
+	                    } else {
+	                        // Try again since some other GWC updated the file in middle of writing process
+	                        ++tryagain;
+	                        log.debug("Reattempting to read metadata file since timestamp changed");
+	                    }
+                    }
+                    
+                } catch (Exception e) {
+                    throw new RuntimeException(e);
+                } finally {
+                    try {
+                        gzipIn.close();
+                        in.close();
+                    } catch (IOException e) {
+                        log.warn(e.getMessage(), e);
+                    }
+                }
+            } else {
+                log.error("Metadata file does not exists: " + metadataFile.getPath());
+            }
+        }
+        return properties;
+    }
+
+    private Properties getLayerMetadata(final String layerName) {
+        final File metadataFile = getMetadataFile(layerName);
+        return this.getLayerMetadata(metadataFile);
+    }
+
+    private String getMetadataFilename(boolean compressedName) {
+        String metadataFilename = "metadata.properties";
+        if (compressedName) {
+            metadataFilename += FileBlobStore.METADATA_GZIP_EXTENSION;
+        }
+        return metadataFilename;
+    }
+
+    /**
+     * Returns the File related to metadata.properties
+     *
+     * @param layerName
+     * @return metadata file (compressed or not, depending if it's present uncompressed)
+     */
     private File getMetadataFile(final String layerName) {
         File layerPath = getLayerPath(layerName);
-        File metadataFile = new File(layerPath, "metadata.properties");
-        return metadataFile;
+        String metadataFilename = this.getMetadataFilename(true);
+        File metadataFile = new File(layerPath, metadataFilename);
+
+        // At this point we backport uncompressed metadata.properties files to compressed version
```

Comment:
This is not a backport, it's for "backwards compatibility".

Reply:
Also, I'd first check if the (compressed) metadata file exists, which would be the new long term "normality", and only when it's missing, see if there is an older metadata file to read and convert.